Add tests for TimeParser

kiwibang opened the pull request:
- Add more tests to the TimeParser class
- Parsing timing is very bug-prone, hence more tests are
needed to ensure the TimeParser API works as intended

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2103T-T11-2/tp

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2103T-T11-2/tp

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2103T-T11-2/tp

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2103T-T11-2/tp

Commit:
Add additional tests for TimeParser
- This adds more tests for the TimeParser class
- Add more tests to check that the API works as intended

Commit:
Remove print statements from test file
- Mistakenly left print statements in the test file
- Production code should not have print statements

Commit:
Fix checkstyle issues
- Removed an unused import which caused CI to fail

JasonLCY-Temp commented on src/test/java/seedu/address/logic/parser/TimeParserTest.java:
```diff
@@ -37,6 +37,97 @@ void testInvalidDate() {
         assertTrue(hasError);
     }
 
+    @Test
+    void testDayAnd24hTime() throws ParseException {
+        TimeParser.parseDate("Tue 1630");
```
Perhaps the test method name here could be something along the lines of `parseDate_validDayAnd24hTime_successfulParse`

kiwibang replied:
Thanks for the heads up! have edited the names of the tests that I am intending to keep! 👍

JasonLCY-Temp reviewed (commented):
Great effort with increasing the test coverage to be more comprehensive.
Just to point out a coding standard that might have been missed in the code:
Test method names should try to follow the three part format `featureUnderTest_testScenario_expectedBehavior()`
with the second and third parts being optional depending on what is being tested exactly 
(parseDate() would contain all the tests for the parseDate method, parseDate_invalidInput() would contain all the test cases with invalid inputs for the parseDate method)

Commit:
Fix method names for tests
- Fix test method name for TimeParser class
- Did not fix test method for some tests,
since I will be deleting them in a new PR
due to changes in parseDate

Commit:
Fix checkstyle issues
- Checkstyle does not allow method underscores,
so I removed them
- The test methods still follow the convention,
just without underscore

Jonyxzx reviewed (approved):
LGTM!
